src/NUnitFramework/framework/Attributes/RandomAttribute.cs

```diff
@@ -258,14 +283,36 @@ public override IEnumerable GetData(IParameterInfo parameter)
 
                 _randomizer = Randomizer.GetRandomizer(parameter.ParameterInfo);
 
+                if (Distinct && _inRange && !CanBeDistinct(_min, _max, _count))
+                    throw new NUnitException($"Random values can't be distinct. Method: \"{parameter.Method}\"; Parameter: \"{parameter.ParameterInfo.Name}\"");
```

Comment:
If no exception is thrown, then there will be an infinite loop when generating random values. the infinite loop can be avoided, checking if this situation occurs.